orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/model/Execution.groovy

```diff
@@ -88,14 +88,14 @@ abstract class Execution<T> implements Serializable {
   }
 
   ExecutionStatus getStatus() {
-    if (canceled) {
-      return CANCELED
-    }
-
     if (version > 1) {
       return executionStatus
     }
 
+    if (canceled) {
```

Comment:
Correct.

Yes, without this chance setting the `canceled` flag always causes the status to be `CANCELED`. Now we're using the flag to cause any cancellation routines to run. This logic is going away soon as you point out.